### PR TITLE
fix: refund cycles attached to provisional_top_up_canister

### PR DESCRIPTION
**Pull request by mraszyk**
This MR specifies that cycles attached to `provisional_top_up_canister` are refunded to the caller.

### CODE REVIEW

**Review by robin-kunzler — APPROVED** [review]
Thanks a lot @mraszyk , LGTM! 👍

File: spec/index.md [hunk 1]
```diff
@@ -2161,6 +2163,8 @@ The signatures are encoded as the concatenation of the [SEC1](https://www.secg.o
 
 This call requires that the ECDSA feature is enabled, the caller is a canister, and `message_hash` is 32 bytes long. Otherwise it will be rejected.
 
+Cycles to pay for the call must be explicitly transferred with the call, i.e., they are not deducted from the caller's balance implicitly (e.g., as for inter-canister calls).
```
Comment (bogwar) [inline spec/index.md:2166]:
```suggestion
Cycles to pay for the call must be explicitly transferred with the call, i.e., they are not automatically deducted from the caller's balance implicitly (e.g., as for inter-canister calls).
```

**Comment by Dfinity-Bjoern** [comment]
Is this behavior what happens in the replica?

**Comment by mraszyk** [comment]
> Is this behavior what happens in the replica?

Yes. See this [line](https://github.com/dfinity/ic/blob/27656c1abce70da733682699af14bfa8c20bba92/rs/execution_environment/src/execution_environment.rs#L1118) or the discussion in the [ticket](https://dfinity.atlassian.net/browse/FOLLOW-671).

**Pull request merged**